src/mpi/coll/transports/gentran/gentran_utils.c

```diff
@@ -319,6 +335,24 @@ void MPII_Genutil_vtx_add_dependencies(MPII_Genutil_sched_t * sched, int vtx_id,
         if (in_vtx->vtx_state != MPII_GENUTIL_VTX_STATE__COMPLETE)
             vtx->pending_dependencies++;
     }
+
+    /* check if there was any fence operation and add appropriate dependencies.
```

Comment:
There needs to be a commit message explaining what purpose sink and fence are implementing. Same as was done for selective sink a few commits ago.

Reply:
Commit message added.

Reply:
Thanks. My next question is why not combine all 3 noop task additions into a single commit? Rather than add one, then add an unrelated testing change, then add 2 more. Makes for a weird series of commits.

Reply:
In fact, why is that bcast testing commit in here at all?

Reply:
Sure, I agree. I will fix the order of commits. The Bcast testing commit should have gone to a  separate pull request - somehow got into this. Sorry about that. Should I leave it in here now (with commit reordering) so that we do not have to rerun tests for it?

Reply:
Ok. All tests are passing. About to push. Will not rerun tests as changes are mere reordering.

Reply:
Done. Pull request updated.